lib/Sema/SemaOverload.cpp

```diff
@@ -12405,6 +12405,10 @@ static FunctionDecl *getBestCandidateForHIP(Sema &S,
     return Cnt0 < Cnt1;
   });
 
+  // Validate number of arguments.  Add 5 for launch configuration
```

Comment:
Shouldn't this check be done in the functor called by std::min_element above?

Reply:
Well, it is a functor for std::min_element., where we're trying to find the kernel function prototype overload that most closely matches the call.  But if the call has an incorrect number of arguments, and we don't support implicit kernel arguments, we should not match the call to the candidate.  I guess, we could make the functor add MAX_INT to the Cnt# so that the functor would cause that candidate to be "max" in the list of candidates?

Reply:
But, we can do the check **before** we call std::min_element, and save us the trouble of checking the argument types against all candidates.

Reply:
yes, for simplicity, if there's a mismatch then we shouldn't even go through the list of args and return false.
But std::min_element would return one candidate at the end, which needs to have a final check like the one below